Frameworks/Foundation/NSProgress.mm

```diff
@@ -200,47 +200,77 @@ - (void)addChild:(NSProgress*)child withPendingUnitCount:(int64_t)inUnitCount {
  @Status Interoperable
 */
 - (void)resignCurrent {
+    auto currentProgressStack = _getProgressStackForCurrentThread();
+
     // If self is the top element on the current progress stack, pop it
     // Be sure to check if self == nil here, otherwise [nil resignCurrent] may cause a pop on an empty stack
```

Comment:
`[nil x]` will never result in a method call. You can rest assured that if you are in a method body, `self != nil`.